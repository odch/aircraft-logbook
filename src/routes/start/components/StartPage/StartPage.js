import React from 'react'
import { Redirect } from 'react-router-dom'
import LoadingIcon from '../../../../components/LoadingIcon'
import { organization } from '../../../../shapes'
import isLoaded from '../../../../util/isLoaded'

class StartPage extends React.Component {
  render() {
    const { selectedOrganization } = this.props
    if (!isLoaded(selectedOrganization)) {
      return <LoadingIcon />
    }
    if (selectedOrganization) {
      return <Redirect to={`/organizations/${selectedOrganization.id}`} />
    }
    return <Redirect to="/organizations" />
  }
}

StartPage.propTypes = {
  selectedOrganization: organization
}

export default StartPage
