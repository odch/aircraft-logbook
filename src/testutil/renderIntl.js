import React from 'react'
import renderer from 'react-test-renderer'
import { createMount } from '@material-ui/core/test-utils'
import { IntlProvider } from 'react-intl'
import pretty from 'pretty'
import messages from '../messages'

const LOCALE = 'de'

const renderInIntlProvider = component => (
  <IntlProvider locale={LOCALE} messages={messages[LOCALE]}>
    {component}
  </IntlProvider>
)

const renderIntl = (component, options) =>
  renderer.create(renderInIntlProvider(component), options)

export const renderIntlUpdate = (renderer, component, options) =>
  renderer.update(renderInIntlProvider(component), options)

export const renderIntlMaterial = (component, returnPrettyHtml = false) => {
  const rendered = createMount()(renderInIntlProvider(component))
  return returnPrettyHtml ? pretty(rendered.html()) : rendered
}

export default renderIntl
