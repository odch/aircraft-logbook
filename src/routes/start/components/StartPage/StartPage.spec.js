import React from 'react'
import { BrowserRouter as Router, Route, Switch } from 'react-router-dom'
import renderIntl from '../../../../testutil/renderIntl'
import StartPage from './StartPage'

const MyOrganizationPage = () => <div>My organization</div>
const OrganizationsPage = () => <div>All organizations</div>

describe('routes', () => {
  describe('start', () => {
    describe('components', () => {
      describe('StartPage', () => {
        it('renders loading icon as long as selectedOrgnaization is not loaded', () => {
          const tree = renderIntl(
            <StartPage selectedOrganization={undefined} />
          ).toJSON()
          expect(tree).toMatchSnapshot()
        })

        it('redirects to selected organization if set', () => {
          const tree = renderIntl(
            <Router>
              <Switch>
                <Route
                  exact
                  path="/organizations/my_org"
                  component={MyOrganizationPage}
                />
                <StartPage selectedOrganization={{ id: 'my_org' }} />
              </Switch>
            </Router>
          ).toJSON()
          expect(tree).toMatchSnapshot()
        })

        it('redirects to organizations page if selected organization not found or set', () => {
          const tree = renderIntl(
            <Router>
              <Switch>
                <Route
                  exact
                  path="/organizations"
                  component={OrganizationsPage}
                />
                <StartPage selectedOrganization={null} />
              </Switch>
            </Router>
          ).toJSON()
          expect(tree).toMatchSnapshot()
        })
      })
    })
  })
})
