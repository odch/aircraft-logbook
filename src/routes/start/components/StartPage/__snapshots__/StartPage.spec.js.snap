// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`routes start components StartPage redirects to organizations page if selected organization not found or set 1`] = `
<div>
  All organizations
</div>
`;

exports[`routes start components StartPage redirects to selected organization if set 1`] = `
<div>
  My organization
</div>
`;

exports[`routes start components StartPage renders loading icon as long as selectedOrgnaization is not loaded 1`] = `
<div
  className="LoadingIcon-container-1"
>
  <div
    className="MuiCircularProgress-root LoadingIcon-icon-2 MuiCircularProgress-colorPrimary MuiCircularProgress-indeterminate"
    role="progressbar"
    style={
      Object {
        "height": 40,
        "width": 40,
      }
    }
  >
    <svg
      className="MuiCircularProgress-svg"
      viewBox="22 22 44 44"
    >
      <circle
        className="MuiCircularProgress-circle MuiCircularProgress-circleIndeterminate"
        cx={44}
        cy={44}
        fill="none"
        r={20.2}
        strokeWidth={3.6}
        style={Object {}}
      />
    </svg>
  </div>
</div>
`;
